src/circleci/rollcage/core.clj

```diff
@@ -137,129 +98,29 @@
-  - The Throwable that was being reported
-  - A map with the result of sending the exception to Rollbar. This map will
-    have the following keys:
-      :err     - an integer, 1 if there was an error sending the exception to
-                 Rollbar, 0 otherwise.
-      :message - A human-readable message describing the error.
-
-  See https://rollbar.com/docs/api/items_post/
-
-  More information on System Properties:
-  https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html"
-  ([access-token]
-   (client access-token {}))
-  ([access-token options]
-   (client* access-token options)))
+  [item]
+  (let [{{:keys [err message]
+          {uuid :uuid} :result} :body} (post endpoint
+                                             {:body (json/generate-string item {:key-fn snake-case})
+                                              :content-type :json
+                                              :as :json})
+        message (format "Item sent successfully err=%s uuid=%s" err uuid)]
+    (println message)
+    (log/log (rollbar-to-logging (:level item)) message)))
+
+(defn ^{:dynamic true} *send* [{:keys [access-token] :as item}]
+  (when-not (string/blank? access-token)
+    (send-item-http item)))
 
 (defn notify
-  ([^String level client ^Throwable exception]
-   (notify level client exception {}))
-  ([^String level {:keys [result-fn  send-fn] :as client} ^Throwable exception {:keys [url params]}]
-   (let [log-level (rollbar-to-logging level)
-         params (merge params (ex-data exception))
-         item (make-rollbar client level exception url params)
-         result (try
-                  (send-fn endpoint exception item)
-                  (catch Exception e
-                    ;; Return an error that matches the shape of the Rollbar API
-                    ;; with an added :exception key
-                    {:err 1
-                     :exception e
-                     :message (.getMessage e)}))]
-     (result-fn exception result)
-     result)))
-
-(defn- report-uncaught-exception
-  [level client exception thread]
-  (notify level client exception
-          {:params {:thread-name (.getName thread)}}))
-
-(defn setup-uncaught-exception-handler
-  "Setup handler to report all uncaught exceptions
-  to rollbar."
-  [client]
-  (Thread/setDefaultUncaughtExceptionHandler
-    (reify Thread$UncaughtExceptionHandler
-      (uncaughtException [_ thread ex]
-        (report-uncaught-exception "critical" client ex thread)))))
-
-(def critical (partial notify "critical"))
-(def error    (partial notify "error"))
-(def warning  (partial notify "warning"))
-(def info     (partial notify "info"))
-(def debug    (partial notify "debug"))
+  [^String level client ^Throwable exception options]
+  (let [item (make-rollbar client level exception options)]
+    (try
+      (*send* item)
+      nil
+      (catch Exception e
+        (log/error e "Failed to send exception to Rollbar")
+        nil))))
```

Comment:
~ I think it would still be useful to return something from this function. Clients can happily ignore it but have the option of taking some action if sending to Rollbar fails.